### ISSUE TITLE
Imap client.connect not giving exception when using bad port number

### ISSUE THREAD

**Issue by Carneno** [jstedfast/MailKit#518]
Hello,

When I use this to connect to an IMAP server using SSL and port 993, it works fine.
`try
               {
                    Mailbox mailbox = db.Mailboxes.Find(mbid);
                    int mailboxcount = db.Mailboxes.Count();
                    using (var client = new ImapClient())
                    {
                         client.ServerCertificateValidationCallback = (s, c, h, e) => true;
                         client.Connect(mailbox.Host, mailbox.ServerPortNumber, mailbox.UseSsl);
                         client.AuthenticationMechanisms.Remove("XOAUTH2");
                         client.Authenticate(mailbox.User, mailbox.Password);
                         var inbox = client.Inbox;
                         inbox.Open(FolderAccess.ReadOnly);
`

If I change the port number to something like 994, it never gets to the **client.AuthenticationMechanisms.Remove("XOAUTH2");** statement.  It does not execute any of the exception catches.  It just exits the method completely.

I used the Visual Studio 2017 debugger and it makes it to the **client.Connect(mailbox.Host, mailbox.ServerPortNumber, mailbox.UseSsl);** line, but when I continue execution, it never gives any kind of error message.

I tried it in IE 11 and Mozilla Firefox.

Is there a way to catch an exception message for this error?

Thanks,
Tony

**Comment by jstedfast** [comment]
Why would you try to connect to port 994?

And yes, it will eventually throw an exception. Same as Socket.Connect() will. It just takes several minutes because of the standard TCP timeout.

**Comment by Carneno** [comment]
Jeffrey,

Thanks for your help.

I myself would not try to connect to port 994, because I know that the correct port to use is 993, but what if one of my clients puts in the wrong port number by mistake when setting up their mailbox?

I want to be able to inform them of what the problem is.

In my testing I did not wait for the timeout to occur.  I just tested it again and waited for the timeout and it did give an error message.

Thanks,
Tony
